docs/source/project_documentation/dev_guide/aac_constraints.md

```diff
@@ -0,0 +1,395 @@
+"""The AaC AaC primitive constraints plugin implementation module."""
+# NOTE: It is safe to edit this file.
+# This file is only initially generated by aac gen-plugin, and it won't be overwritten if the file already exists.
+
+# There may be some unused imports depending on the definition of the plugin...but that's ok
+from aac.execute.aac_execution_result import (
+    ExecutionResult,
+    ExecutionStatus,
+    ExecutionMessage,
+    MessageLevel,
+)
+
+from aac.context.language_context import LanguageContext
+from aac.context.definition import Definition
+from aac.context.source_location import SourceLocation
+from typing import Any
+
+
+plugin_name = "AaC primitive constraints"
+
+
+def check_dataref(
+    value: str, type_declaration: str, source: AaCFile, location: SourceLocation
+) -> ExecutionResult:
+    """Business logic for the Check dataref constraint."""
+
+    # TODO: rewrite the code below to implement your constraint logic
+    # the checker has provided you with source and location data for messages
+    status = ExecutionStatus.GENERAL_FAILURE
+    messages: list[ExecutionMessage] = []
+    error_msg = ExecutionMessage(
+        message="The Check dataref constraint for the AaC primitive constraints plugin has not been implemented yet.",
+        level=MessageLevel.ERROR,
+        source=source,
+        location=location,
+    )
+    messages.append(error_msg)
+
+    return ExecutionResult(plugin_name, "Check dataref", status, messages)
+
+```
+
+Generated unit test stub:
+
+```yaml
+from unittest import TestCase
+from typing import Tuple
+from click.testing import CliRunner
+from aac.execute.command_line import cli, initialize_cli
+from aac.execute.aac_execution_result import ExecutionStatus
+
+
+from aac.plugins.aac_primitives.aac_primitive_constraints_impl import plugin_name
+
+from aac.plugins.aac_primitives.aac_primitive_constraints_impl import check_dataref
+
+
+class TestAaCprimitiveconstraints(TestCase):
+    def test_check_dataref(self):
+        # TODO: Write tests for check_dataref
+        self.fail("Test not yet implemented.")
+
+```
+
+You may now create your primitive constraint test and implementation.  emember to add an empty `__init__.py` to your test director and run `aac clean` to remove any extra files if needed.
+
+Finally, be sure to assign the primitive constraint you've created to your primitive declaration.
+
+```yaml
+primitive:
+  name: dataref
+  package: aac.lang
+  python_type: str
+  description:  |
+    A reference to another defined data item.  References are qualified and documented 
+    as 'reference(schema_root.field_name)' denoting the definition and field being referenced.
+    Do not confuse data references with type references.  A data reference can be thought of as
+    a pointer to a specific data item, while a type reference is a pointer to a definition.  Data
+    references do not comprehend type inheritance or polymorphism.
+  constraints:
+    - name: Check dataref
+```
```

Comment:
This can be replaced with the literal include capability provided by Sphinx.

```suggestion
_aac.aac_
```{eval-rst}
.. literalinclude:: ../../../../python/src/aac/aac.aac
    :language: yaml
    :lines: 173-184
```

There should be a second set of the three ` to close out the literal include, but it's rendering oddly.